<launch>

  <arg name="namespace" default="/" />
  <arg name="respawn" default="False" />
  <arg name="gui" default="False" />
  <arg name="osi_pi" default="False" />
  <arg name="use_proxy" default="False" />

  <!-- launch diagnotsics aggregator node -->
  <include file="$(find smelter_diagnostics)/launch/diagnostics_aggregator_node.launch">
    <arg name="namespace" value="$(arg namespace)" />
    <arg name="respawn" value="$(arg respawn)" />
    <arg name="gui" value="$(arg gui)" />
  </include>

  <!-- launch diagnostics publisher node -->
  <include
      if="$(arg osi_pi)"
      file="$(find smelter_diagnostics)/launch/diagnostics_publisher_node.launch">

    <arg name="namespace" value="$(arg namespace)" />
    <arg name="respawn" value="$(arg respawn)" />
    <arg name="use_proxy" value="$(arg use_proxy)" />
  </include>

</launch>
